### PR TITLE
Fix flaky tests and unskip

### PR DESCRIPTION
**Pull request by grouma**
Closes https://github.com/dart-lang/webdev/issues/924

Related: https://github.com/dart-lang/test/issues/1203

### CODE REVIEW

**Comment by grouma** [comment]
@natebosch 

This somehow starves the test runner when the event doesn't come through. We do see an error message on `webdev.stdout`. Any idea what's going on?

```Dart
await expectLater(
            webdev.stdout,
            emitsThrough(
                startsWith('[{"event":"app.log","params":{"appId":"$appId",'
                    '"log":"Hello World\\n"}}')));
```

**Comment by natebosch** [comment]
I don't have context, are you asking if I know why we get an error message on `webdev.stdout`?

**Comment by grouma** [comment]
No. I'm asking how the `expectLater`, `TestProcess` `stdout`, and or the `emitsThrough` could starve the test runner.

File: webdev/pubspec.yaml [hunk 1]
```diff
@@ -45,4 +45,8 @@ dev_dependencies:
   webdriver: ^2.0.0
 
 executables:
-  webdev:
\ No newline at end of file
+  webdev:
+
+dependency_overrides:
+  dwds:
+    path: ../dwds
```
Comment (natebosch) [inline webdev/pubspec.yaml]: Missing trailing newline

File: webdev/test/daemon/app_domain_test.dart [hunk 2]
```diff
@@ -62,6 +63,11 @@ void main() {
         var webdev =
             await runWebDev(['daemon'], workingDirectory: exampleDirectory);
         var appId = await waitForAppId(webdev);
+        if (Platform.isWindows) {
+          // Windows takes a bit longer to run the application and register
+          // the service extension.
+          await Future.delayed(const Duration(seconds: 5));
```
Comment (natebosch) [inline webdev/test/daemon/app_domain_test.dart]: Gross...

Can you file an issue to get rid of the arbitrary wait?

**Pull request merged**